src/goto-programs/process_goto_program.cpp

```diff
@@ -58,9 +58,16 @@ bool process_goto_program(
   }
 
   // remove returns, gcc vectors, complex
-  remove_returns(goto_model);
+  if(
+    options.get_bool_option("remove-returns") ||
+    options.get_bool_option("string-abstraction"))
```

Comment:
String abstraction was built well before return-value removal, so I wouldn't really expect a dependency here. In 425731b66727c40607457cef1a82acb3f6f316b6 I noticed a need for some fixes, but I don't think that there should be a general requirement for return-value removal to make string abstraction work.

Reply:
It's unclear what the effort is. Might not be worth it, given that this feature has no known users.